packages/sdk/src/clients/bridgePool/client.ts

```diff
@@ -0,0 +1,167 @@
+import { BridgePoolEthers, BridgePoolEthers__factory } from "@uma/contracts-node";
+import type { SignerOrProvider, GetEventType } from "../..";
+import { Event, BigNumber } from "ethers";
+import { Balances } from "../../utils";
+
+export type Instance = BridgePoolEthers;
+const Factory = BridgePoolEthers__factory;
+
+export function connect(address: string, provider: SignerOrProvider): Instance {
+  return Factory.connect(address, provider);
+}
+
+export type LiquidityAdded = GetEventType<Instance, "LiquidityAdded">;
+export type LiquidityRemoved = GetEventType<Instance, "LiquidityRemoved">;
+export type DepositRelayed = GetEventType<Instance, "DepositRelayed">;
+export type RelaySpedUp = GetEventType<Instance, "RelaySpedUp">;
+export type RelayDisputed = GetEventType<Instance, "RelayDisputed">;
+export type RelayCanceled = GetEventType<Instance, "RelayCanceled">;
+export type RelaySettled = GetEventType<Instance, "RelaySettled">;
+export type BridgePoolAdminTransferred = GetEventType<Instance, "BridgePoolAdminTransferred">;
+
+export type DepositData = {
+  chainId: number;
+  depositId: BigNumber;
+  l1Recipient: string;
+  l2Sender: string;
+  amount: BigNumber;
+  slowRelayFeePct: BigNumber;
+  instantRelayFeePct: BigNumber;
+  quoteTimestamp: number;
+};
+export type RelayData = {
+  relayState: number;
+  slowRelayer: string;
+  relayId: number;
+  realizedLpFeePct: BigNumber;
+  priceRequestTime: number;
+  proposerBond: BigNumber;
+  finalFee: BigNumber;
+};
+export interface EventState {
+  oldAdmin?: string;
+  newAdmin?: string;
+  tokens: Record<string, Balances>;
+  lpTokens: Record<string, Balances>;
+  deposits: Record<string, DepositData>;
+  relays: Record<string, RelayData>;
+  instantRelays: Record<string, string>;
+}
+
+function eventStateDefaults() {
+  return {
+    tokens: {},
+    lpTokens: {},
+    deposits: {},
+    relays: {},
+    instantRelays: {},
+  };
+}
+
+export function reduceEvents(state: EventState, event: Event): EventState {
+  switch (event.event) {
+    // event LiquidityAdded(address indexed token, uint256 amount, uint256 lpTokensMinted, address liquidityProvider);
+    case "LiquidityAdded": {
+      const typedEvent = event as LiquidityAdded;
+      const { token, amount, lpTokensMinted, liquidityProvider } = typedEvent.args;
+      const tokens = Balances(state.tokens[token] || {});
+      const lpTokens = Balances(state.lpTokens[token] || {});
```

Comment:
I am assuming that token balances and lp token balances are distinct for each token address